internal/translator/data/rules_test.go

```diff
@@ -0,0 +1,232 @@
+					Tag:  "ADJ",
+					Type: "Q",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 2).Lemma}}",
+				},
+				{
+					"type":  "literal",
+					"value": " ",
+				},
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 1).Lemma}}",
+				},
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		newID, error := helper.NewRule(context.Background(), newRule)
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("New rule created with id %s", newID.Hex())
+	})
+
+	t.Run("save a new rule with success and with the same last pattern but different detail when the strucutre is valid", func(t *testing.T) {
+		newRule := Rule{
+			SourceLanguage: "espaol",
+			TargetLanguage: "kaqchikel",
+			Pattern:        "VERB,ADV,ADJ",
+			Details: []RuleDetail{
+				{
+					Tag:  "VERB",
+					Type: "M",
+				},
+				{
+					Tag:  "ADV",
+					Type: "G",
+				},
+				{
+					Tag:  "ADJ",
+					Type: "O",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 1).Lemma}}",
+				},
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		newID, error := helper.NewRule(context.Background(), newRule)
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("New rule created with id %s", newID.Hex())
+	})
+
+	t.Run("find rule with success when the value exists", func(t *testing.T) {
+		rule, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "VERB,ADV,ADJ")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("Found rules %v", rule)
+	})
+
+	t.Run("get and error finding non existing rule", func(t *testing.T) {
+		rule, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "ADJ,ADJ,ADV")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		if len(rule) == 0 {
+			t.Logf("pattern ADJ,ADJ,ADV doesn't exist")
+		}
+	})
+
+	t.Run("delete a group of rules with success when the rules exist", func(t *testing.T) {
+		error := helper.DeleteMany(context.Background(), "espaol", "kaqchikel", "VERB,ADV,ADJ")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("Rules deleted")
+	})
+
+	t.Run("add short new rule with success when the strucutre is valid", func(t *testing.T) {
+		newRule := Rule{
+			SourceLanguage: "espaol",
+			TargetLanguage: "kaqchikel",
+			Pattern:        "VERB",
+			Details: []RuleDetail{
+				{
+					Tag:  "VERB",
+					Type: "M",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		newID, error := helper.NewRule(context.Background(), newRule)
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("New rule created with id %s", newID.Hex())
+	})
+
+	t.Run("update one rule with success when the rule exists", func(t *testing.T) {
```

Comment:
What if the rule does not exists? Missing test

Reply:
the array is length 0, and the test was written , check the test again and you will find it

Reply:
Which one is the test? This one says `update one rule with success when the rule exists`

Where is the `get an error updating a non existent rule`?

What i'm pointing is that you need to test errors as well.

Reply:
What do you think about this?